camp/camp-brooklyn/src/main/java/org/apache/brooklyn/camp/brooklyn/spi/creation/BrooklynEntityDecorationResolver.java

```diff
@@ -180,20 +182,23 @@ protected void addDecorationFromJsonMap(Map<?, ?> decorationJson, List<EntityIni
     // Not much value from extending from BrooklynEntityDecorationResolver, but let's not break the convention
     public static class SpecParameterResolver extends BrooklynEntityDecorationResolver<SpecParameter<?>> {
 
+        private Function<Object, Object> transformer;
+
         protected SpecParameterResolver(BrooklynYamlTypeInstantiator.Factory instantiator) { super(instantiator); }
 
         @Override
         protected String getDecorationKind() { return "Spec Parameter initializer"; }
 
         @Override
-        public void decorate(EntitySpec<?> entitySpec, ConfigBag attrs) {
+        public void decorate(EntitySpec<?> entitySpec, ConfigBag attrs, Set<String> encounteredRegisteredTypeIds) {
+            transformer = new BrooklynComponentTemplateResolver.SpecialFlagsTransformer(instantiator.loader, encounteredRegisteredTypeIds);
```

Comment:
Would it work if you transform `attrs` before passing it to `BasicSpecParameter.fromConfigList` so it doesn't need to know about the `transformer`?